trl/trainer/ppo_trainer.py

```diff
@@ -265,17 +265,43 @@ def __init__(
         else:
             self.kl_ctl = FixedKLController(self.config.init_kl_coef)
 
-        (
-            self.model,
-            self.ref_model,
-            self.optimizer,
-            self.data_collator,
-            self.dataloader,
-            self.lr_scheduler,
-        ) = self.accelerator.prepare(
-            self.model, self.ref_model, self.optimizer, self.data_collator, self.dataloader, self.lr_scheduler
+        # Safety checkers for DS integration
+        is_deepspeed_used = self.accelerator.distributed_type == "DEEPSPEED" and hasattr(
+            self.accelerator.state, "deepspeed_plugin"
         )
 
+        if is_deepspeed_used:
+            (
+                self.model,
+                self.optimizer,
+                self.data_collator,
+                self.dataloader,
+                self.lr_scheduler,
+            ) = self.accelerator.prepare(
+                self.model, self.optimizer, self.data_collator, self.dataloader, self.lr_scheduler
+            )
```

Comment:
we could move this block outside the if-else and then just call `self.ref_model = self.accelerator.prepare(self.ref_model)` in the else clause, no?